File: corpus/typescript/fixture_25.ts
type CartItem25 = { price: number; qty?: number };
export function calculateDiscount25(cartItems: CartItem25[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 250 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook25 {
  lookupPrice(sku: string): number { return sku.length + 25; }
}

File: corpus/rust/fixture_25.rs
pub struct PriceBook25 {
    pub multiplier: f64,
}

pub fn calculate_discount_25(cart_items: &[(f64, f64)], tax_rate: f64) -> f64 {
    let total_amount: f64 = cart_items.iter().map(|(price, qty)| price * qty).sum();
    if total_amount > 250.0 { total_amount * (1.0 + tax_rate) } else { total_amount }
}
